ChatRPG/Services/GameInputHandler.cs

```diff
@@ -60,81 +66,118 @@ private void OnChatCompletionChunkReceived(bool isStreamingDone, string? chunk =
                 }
-                messageContent += $"The player hits with their attack, dealing {playerDmg} damage. The opponent has {opponent.CurrentHealth} health remaining.";
-                _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", campaign.Player.Name, opponent.Name, playerDmg, opponent.CurrentHealth, opponent.MaxHealth);
-            }
-            else
-            {
-                messageContent += $"The player misses with their attack, dealing no damage. The opponent has {opponent.CurrentHealth} health remaining.";
-            }
 
-            if (opponentDmg != 0)
-            {
-                messageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has {campaign.Player.CurrentHealth} health remaining.";
-                if (campaign.Player.AdjustHealth(-opponentDmg))
+                if (opponentDmg != 0)
                 {
-                    await HandlePlayerDeath(campaign.Player, conversation);
+                    combatMessageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has {(campaign.Player.CurrentHealth - opponentDmg < 0 ? 0 : campaign.Player.CurrentHealth - opponentDmg)} health remaining.";
+                    if (campaign.Player.AdjustHealth(-opponentDmg))
+                    {
+                        combatMessageContent += "The player has died and their adventure ends.";
+                    }
+                    _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", opponent.Name, campaign.Player.Name, opponentDmg, campaign.Player.CurrentHealth, campaign.Player.MaxHealth);
                 }
-                _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", opponent.Name, campaign.Player.Name, opponentDmg, campaign.Player.CurrentHealth, campaign.Player.MaxHealth);
-            }
-            else
+                else
+                {
+                    combatMessageContent += $"The opponent will miss their next attack, dealing no damage. The player has {campaign.Player.CurrentHealth} health remaining.";
+                }
+
+                OpenAiGptMessage combatSystemMessage = new(ChatMessageRole.System, combatMessageContent);
+                conversation.Add(combatSystemMessage);
+                return _systemPrompts[systemPromptType];
+            default:
+                throw new ArgumentOutOfRangeException();
+        }
+    }
+
+    private async void DetermineAndPerformHurtOrHeal(Campaign campaign, ICollection<OpenAiGptMessage> conversation)
+    {
+        OpenAiGptMessage lastUserMessage = conversation.Last(m => m.Role.Equals(ChatMessageRole.User));
+        string hurtOrHealString = await _llmClient.GetChatCompletion(new List<OpenAiGptMessage>() { lastUserMessage }, _systemPrompts[SystemPromptType.HurtOrHeal]);
```

Comment:
Flasher lige noget Java goodness hehe
```suggestion
        string hurtOrHealString = await _llmClient.GetChatCompletion(List.of(lastUserMessage), _systemPrompts[SystemPromptType.HurtOrHeal]);
```
(dont commit this)

Reply:
ok bro